docs/guide/images.md

```diff
@@ -398,3 +398,171 @@ For an ImageAsset Field to be reused you have to publish it. Only then is notifi
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentValue
+     * @throws \eZ\Publish\Core\Base\Exceptions\NotFoundException
+     */
+    public function downloadSvgAction(
+        int $contentId,
+        string $fieldIdentifier,
+        string $filename,
+        Request $request
+    ): Response {
+        $version = null;
+
+        if ($request->query->has('version')) {
+            $version = $request->query->get('version');
+        }
+
+        $content = $this->contentService->loadContent($contentId, null, $version);
+        $language = $request->query->has('inLanguage') ? $request->query->get('inLanguage') : null;
+        $field = $this->translationHelper->getTranslatedField($content, $fieldIdentifier, $language);
+
+        if (!$field instanceof Field) {
+            throw new InvalidArgumentException(sprintf("%s field not present in content %d '%s'", $fieldIdentifier, $content->contentInfo->id, $content->contentInfo->name));
+        }
+
+        $binaryFile = $this->ioService->loadBinaryFile($field->value->id);
+        $response = new Response($this->ioService->getFileContents($binaryFile));
+        $disposition = $response->headers->makeDisposition(
+            ResponseHeaderBag::DISPOSITION_INLINE, $filename
+        );
+
+        $response->headers->set('Content-Disposition', $disposition);
+        $response->headers->set('Content-Type', self::CONTENT_TYPE_HEADER);
+
+        return $response;
+    }
+}
+```
+
+To be able to use a proper link in your templates, you also need a dedicated Twig extension:
+
+```php
+<?php
+
+namespace App\Twig;
+
+use Symfony\Component\Routing\RouterInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class SvgExtension extends AbstractExtension
+{
+    /** @var \Symfony\Component\Routing\RouterInterface */
+    protected $router;
+
+    /**
+     * SvgExtension constructor.
+     */
+    public function __construct(RouterInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * @return array|TwigFunction[]
+     */
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('ibexa_svg_link', [
+                $this,
+                'generateLink',
+            ]),
+        ];
+    }
+
+    public function generateLink(int $contentId, string $fieldIdentifier, string $filename): string
+    {
+        return $this->router->generate('app.svg_download', [
+            'contentId' => $contentId,
+            'fieldIdentifier' => $fieldIdentifier,
+            'filename' => $filename,
+        ]);
+    }
+}
+```
+
+Next, you configure the extension within the `config/services.yaml` file:
+
+```yaml
+App\Twig\SvgExtension:
+    arguments:
+        - "@router"
+    tags:
+        - { name: twig.extension }
```

Comment:
SF4+:
```suggestion
App\Twig\SvgExtension:
    autowire: true
    autoconfigure: true
```
^ note: you still might need `'@router'` argument if it's not autowirable, but Twig extension definitely needs to be autoconfigured - check that please.

Reply:
I checked, marking the extension's definition as autowired and autoconfigured is enough in this case.